tools/failures/detect_new_failures.py

```diff
@@ -0,0 +1,259 @@
+#!/usr/bin/env python
```

Comment:
nit: I'm not a bit fan of the tools/failures naming (it's yet another new directory under tools/ which is already messy enough), but if you think that's a good name then let's use it (and we can move scripts around later).